projects/storefrontlib/src/events/page/page-event.builder.ts

```diff
@@ -1,58 +1,55 @@
 import { Injectable } from '@angular/core';
-import { ofType } from '@ngrx/effects';
-import { RouterNavigatedAction, ROUTER_NAVIGATED } from '@ngrx/router-store';
-import { ActionsSubject } from '@ngrx/store';
 import {
-  ActivatedRouterStateSnapshot,
   createFrom,
   EventService,
+  FeatureConfigService,
 } from '@spartacus/core';
 import { Observable } from 'rxjs';
 import { filter, map } from 'rxjs/operators';
+import { NavigationEvent } from '../navigation/navigation.event';
 import { HomePageEvent, PageEvent } from './page.events';
 
 @Injectable({
   providedIn: 'root',
 })
 export class PageEventBuilder {
   constructor(
-    protected actions: ActionsSubject,
-    protected eventService: EventService
+    protected eventService: EventService,
+    /** @deprecated @since 3.1 - this will be remove in 4.0 */ protected featureConfigService?: FeatureConfigService
   ) {
     this.register();
   }
 
   protected register(): void {
-    this.eventService.register(PageEvent, this.buildPageEvent());
+    if (this.featureConfigService?.isLevel('!3.1')) {
+      this.eventService.register(PageEvent, this.buildPageEvent());
```

Comment:
Wouldn't it be more clear to have separate HomePageEventBuilder, so this big one would be deprecated and removed in 4.0?

Reply:
Done